pyof/foundation/network_types.py

```diff
@@ -470,7 +475,7 @@ def __init__(self, tlv_type=1, sub_type=7, sub_value=BinaryData()):
         """
         super().__init__(tlv_type)
         self.sub_type = sub_type
-        self.sub_value = sub_value
+        self.sub_value = BinaryData() if sub_value is None else sub_value
```

Comment:
The cleaner way is to set `BinaryData()` directly in the `__init__` method - same behavior, better readability.